### PR TITLE
feat(replay): Add information about adding CSP entry for Replay WebWorkers

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

**Comment by mydea** [comment]
I think this is good, but IMHO we should maybe call this out more aggressively in the setup. Maybe add this as well somewhere here: https://docs.sentry.io/platforms/javascript/session-replay/#set-up ?

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 1]
```diff
@@ -39,6 +39,13 @@ By default, it'll mask all fields, so no personally identifiable information (PI
 
 <PlatformContent includePath="session-replay/setup" />
 
+### Content Security Policy (CSP)
+Session Replay uses a WebWorker to perform work off the main UI thread as to not degrade performance of your application. Ensure that workers can be loaded from our CDN by adding this entry:
+
```
Comment (lizokm) [inline src/platforms/javascript/common/session-replay/index.mdx:44]:
```suggestion
Session Replay uses a WebWorker to perform work off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded from our CDN:


```

**Review by lizokm — APPROVED** [review]
Just a few small tweaks :)

**Comment by billyvg** [comment]
@mydea moved to setup section

**Review by mydea — APPROVED** [review]

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 2]
```diff
@@ -39,6 +39,14 @@ By default, it'll mask all fields, so no personally identifiable information (PI
 
 <PlatformContent includePath="session-replay/setup" />
 
+### Content Security Policy (CSP)
+Session Replay uses a WebWorker to perform work off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded from our CDN:
```
Comment (billyvg) [inline src/platforms/javascript/common/session-replay/index.mdx:43]: Unfortunately, we can't load cross origin web workers :(

```suggestion
Session Replay uses a WebWorker to perform work (e.g. compression) off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded: 
```

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 3]
```diff
@@ -39,6 +39,14 @@ By default, it'll mask all fields, so no personally identifiable information (PI
...
+
+
+```
+worker-src https://browser.sentry-cdn.com
+```
+
```
Comment (billyvg) [inline src/platforms/javascript/common/session-replay/index.mdx:49]:
```suggestion

Older versions of Safari do not support `worker-src`, you will need to additionally add an entry for `child-src`:

child-src 'self' blob:
```

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 4]
```diff
@@ -39,6 +39,14 @@ By default, it'll mask all fields, so no personally identifiable information (PI
 
 <PlatformContent includePath="session-replay/setup" />
 
+### Content Security Policy (CSP)
+Session Replay uses a WebWorker to perform work off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded from our CDN:
+
+
+```
+worker-src https://browser.sentry-cdn.com
```
Comment (billyvg) [inline src/platforms/javascript/common/session-replay/index.mdx:47]:
```suggestion
worker-src 'self' blob:
```

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 5]
```diff
@@ -39,6 +39,17 @@ By default, it'll mask all fields, so no personally identifiable information (PI
...
+Session Replay uses a WebWorker to perform work (e.g. compression) off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded: 
+
+
+```
+worker-src 'self' blob:
+```
+
+Older versions of Safari do not support `worker-src`, you will need to additionally add an entry for `child-src`:
+
+child-src 'self' blob:
```
Comment (billyvg) [inline src/platforms/javascript/common/session-replay/index.mdx:52]:
```suggestion
```
child-src 'self' blob:
```

```

**Review by mydea — APPROVED** [review]
nice 👍

File: src/platforms/javascript/common/session-replay/index.mdx [hunk 6]
```diff
@@ -39,6 +39,20 @@ By default, it'll mask all fields, so no personally identifiable information (PI
...
 
+### Content Security Policy (CSP)
+Session Replay uses a WebWorker to perform work (e.g. compression) off the main UI thread so as not to degrade the performance of your application. Add the below entry to make sure that workers can be loaded: 
+
+
+```
+worker-src 'self' blob:
+```
+
+Older versions of Safari do not support `worker-src`, you will need to additionally add an entry for `child-src`:
```
Comment (bruno-garcia) [inline src/platforms/javascript/common/session-replay/index.mdx:50]: Do we know what versions exactly? As a reader I'm confused: Well I have users on Safari so do I need to add this too?

Comment (billyvg) [inline src/platforms/javascript/common/session-replay/index.mdx:50]:
```suggestion
[Safari versions <= 15.4](https://caniuse.com/?search=worker-src) do not support `worker-src`, you will need to additionally add an entry for `child-src`:
```

**Pull request merged**